sdk/servicebus/Azure.Messaging.ServiceBus/tests/Receiver/ReceiverLiveTests.cs

```diff
@@ -46,6 +47,44 @@ await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitionin
             }
         }
 
+        [Test]
+        public async Task PeekWithZeroTimeout()
+        {
+            await using (var scope =
+                await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: false))
+            {
+                await using var client = new ServiceBusClient(TestEnvironment.ServiceBusConnectionString);
+                await using var receiverWithPrefetch = client.CreateReceiver(scope.QueueName,
+                    options: new ServiceBusReceiverOptions() {PrefetchCount = 10});
+                await using var receiver = client.CreateReceiver(scope.QueueName);
+
+                // Test without prefetch
+                var stopwatch = new Stopwatch();
+                stopwatch.Start();
+                await receiver.ReceiveMessagesAsync(10, TimeSpan.Zero).ConfigureAwait(false);
+
+                stopwatch.Stop();
+                var durationInSecs = stopwatch.Elapsed.TotalSeconds;
+
+                // Test with prefetch
+                stopwatch.Reset();
+                stopwatch.Start();
+                await receiverWithPrefetch.ReceiveMessagesAsync(10, TimeSpan.Zero).ConfigureAwait(false);
+                stopwatch.Stop();
+                var durationWithPrefetchModeInSecs = stopwatch.Elapsed.TotalSeconds;
+
+                // Asserts:
+                Assert.IsTrue(durationWithPrefetchModeInSecs < durationInSecs);
+
+                // If prefetch isn't enabled, timeout 0 secs will be replaced with 10 seconds.
+                Assert.IsTrue(durationInSecs > 10);
```

Comment:
Isn't this testing an implementation detail in the AMQP library?  Maybe we should be looser in criteria here and just assert that we're seeing `durationInSecs >= 1`?

Reply:
It was a good point. But I removed that part of test because of Josh's feedback "not allow zero as timeout without prefetch enabled".